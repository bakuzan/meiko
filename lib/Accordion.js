import classNames from 'classnames';
import React, { useState, useEffect } from 'react';

import { usePrevious } from './hooks/usePrevious';
import styles from './styles/Accordion';

function AccordionToggle({ className, heading, onToggle }) {
  return (
    <button type="button" className={className} onClick={onToggle}>
      {heading}
    </button>
  );
}

function Accordion({
  children,
  className,
  contentProps = {},
  defaultIsCollapsed,
  headingProps = {},
  heading,
  toggleComponent = AccordionToggle,
  onToggle,
  ...pass
}) {
  const [isCollapsed, setCollapsed] = useState(defaultIsCollapsed ?? true);
  const prevCollapsed = usePrevious(isCollapsed);

  const Toggler = toggleComponent;

  useEffect(() => {
    const hasPrev = prevCollapsed !== undefined;
    const hasNew = prevCollapsed !== isCollapsed;

    if (onToggle && hasPrev && hasNew) {
      onToggle(isCollapsed);
    }
  }, [onToggle, isCollapsed, prevCollapsed]);

  return (
    <div
      {...pass}
      className={classNames('accordion', styles.accordion, className)}
    >
      <div
        {...headingProps}
        className={classNames(
          'accordion__heading',
          styles.accordion__heading,
          headingProps.className
        )}
      >
        <Toggler
          className={classNames(
            'accordion__toggle',
            styles.accordion__toggle,
            isCollapsed && [
              'accordion__toggle--checked',
              styles.accordion__toggle_collapsed
            ]
          )}
          heading={heading}
          onToggle={() => setCollapsed((p) => !p)}
        />
      </div>
      <div
        {...contentProps}
        aria-hidden={isCollapsed}
        className={classNames(
          'accordion__content',
          styles.accordion__content,
          contentProps.className,
          isCollapsed && [
            'accordion__content--collapsed',
            styles.accordion__content_collapsed
          ]
        )}
      >
        {typeof children !== 'function' ? children : children(isCollapsed)}
      </div>
    </div>
  );
}

Accordion.propTypes = {
  heading: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node
  ]).isRequired,
  defaultIsCollapsed: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node
  ]),
  onToggle: PropTypes.func
};

export default Accordion;
