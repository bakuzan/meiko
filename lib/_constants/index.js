export { default as Strings } from './strings';
export { default as Types } from './types';
export { default as Urls } from './urls';
export { default as Icons } from './icons';
export { PositionEnum, KeyCodes, OPEN_KEYS, CLOSE_KEYS } from './enums';
