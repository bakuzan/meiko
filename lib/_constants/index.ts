import _Strings from './strings';
import _Types from './types';
import _Urls from './urls';
import _Icons from './icons';
import * as _Enums from './enums';

export const Strings = _Strings;
export const Types = _Types;
export const Urls = _Urls;
export const Icons = _Icons;
export const Enums = _Enums;
