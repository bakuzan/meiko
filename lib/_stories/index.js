import React from 'react';
// import { storiesOf } from '@storybook/react';

import './styles/index.scss';

// storiesOf('Index', module).add('basic', () => <div>Exists to load styles</div>);

export const storyStyle = {
  maxWidth: '300px',
  margin: 'auto',
  marginTop: '20px'
};

const styles = { display: 'flex', justifyContent: 'center', marginTop: '30px' };
export const CenterDecorator = (storyFn) => (
  <div style={styles}>{storyFn()}</div>
);
