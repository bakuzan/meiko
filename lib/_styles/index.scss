@import './partials/variables.scss';
@import './box-model.scss';
@import './list.scss';
@import './input-controls.scss';

.button-group {
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 5px;
  margin: 5px 0;

  &.centered {
    justify-content: center;
  }
  &.right-aligned {
    justify-content: flex-end;
  }
}
