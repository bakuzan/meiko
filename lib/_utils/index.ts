import adjustDateMonth from 'ayaka/adjustDateMonth';
import adjustDateYear from 'ayaka/adjustDateYear';
import { capitalise } from 'ayaka/capitalise';
import checkIfDatePartsMatch from 'ayaka/checkIfDatePartsMatch';
import convertToBase64 from 'ayaka/convertToBase64';
import createListeners from 'ayaka/createListeners';
import debounce from 'ayaka/debounce';
import formatDateForInput from 'ayaka/formatDateForInput';
import generateUniqueId from 'ayaka/generateUniqueId';
import getDaysInMonthForDate from 'ayaka/getDaysInMonthForDate';
import getDifferenceFromMonday from 'ayaka/daysDifferenceFromMonday';
import getDifferenceFromSunday from 'ayaka/daysDifferenceFromSunday';
import getElementCoordinates from 'ayaka/getElementCoordinates';
import getEventValue from 'ayaka/getEventValue';
import getFirstDateOfMonth from 'ayaka/getFirstDateOfMonth';
import getLastDateOfMonth from 'ayaka/getLastDateOfMonth';
import getMonthName from 'ayaka/getMonthName';
import { getTimeoutSeconds } from 'ayaka/getTimeout';
import isArray from 'ayaka/isArray';
import { isBefore, isBeforeOrEqual } from 'ayaka/isBefore';
import isNumber from 'ayaka/isNumber';
import isObject from 'ayaka/isObject';
import isString from 'ayaka/isString';
import { lighten, darken } from 'ayaka/palette';

export {
  adjustDateMonth,
  adjustDateYear,
  capitalise,
  convertToBase64,
  createListeners,
  debounce,
  generateUniqueId,
  getElementCoordinates,
  getEventValue,
  getTimeoutSeconds,
  isArray,
  isNumber,
  isObject,
  isString,
  checkIfDatePartsMatch,
  formatDateForInput,
  getMonthName,
  getDaysInMonthForDate,
  getFirstDateOfMonth,
  getLastDateOfMonth,
  getDifferenceFromMonday,
  getDifferenceFromSunday,
  isBefore,
  isBeforeOrEqual,
  lighten,
  darken
};
