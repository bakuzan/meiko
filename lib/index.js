// Components
export { default as Alert } from './Alert';
export { default as AppInformation } from './AppInformation';
export { default as AutocompleteInput } from './AutocompleteInput';
export { Button, withButtonisation } from './Button';
export { default as ChipListInput } from './ChipListInput';
export { default as ClearableInput } from './ClearableInput';
export { default as DateSelector } from './DateSelector';
export { default as Dialog } from './Dialog';
export { default as DropdownMenu } from './DropdownMenu';
export { default as FileUploader } from './FileUploader';
export { default as Form } from './Form';
export { default as FormControls } from './FormControls';
export { default as Grid } from './Grid';
export { default as Header } from './Header';
export { default as Image } from './Image';
export { default as ImageSelector } from './ImageSelector';
export { default as List } from './List';
export { SimpleLoading, Loading } from './Loadable';
export { default as LoadableContent } from './LoadableContent';
export { default as LoadingBouncer } from './LoadingBouncer';
export { default as LoadingSpinner } from './LoadingSpinner';
export { default as Logo } from './Logo';

export { default as MultiSelect } from './MultiSelect';
export { default as RadioButton } from './RadioButton';
export { default as RadioToggle } from './RadioToggle';
export { default as RatingControl } from './RatingControl';
export { default as RequestIndicator } from './RequestIndicator';
export { default as ScrollTopButton } from './ScrollTopButton';
export { default as SelectBox } from './SelectBox';
export { default as Sidebar } from './Sidebar';
export { default as Tabs } from './Tabs';
export { default as TagCloudSelector } from './TagCloudSelector';
export { default as Tickbox } from './Tickbox';
export { default as Toaster } from './Toaster';
export { default as Tooltip } from './Tooltip';

export { default as NewTabLink } from './NewTabLink';
export { default as Portal } from './Portal';

// Hooks
export { useGlobalStyles } from './hooks/useGlobalStyles';

// Utils
export { default as toaster, ToasterService } from './utils/toasterService';
export { default as addOutsideClick } from './utils/addOutsideClick';

export {
  PositionEnum,
  KeyCodes,
  OPEN_KEYS,
  CLOSE_KEYS,
  Icons,
  Strings,
  Types,
  Urls
} from './constants';
