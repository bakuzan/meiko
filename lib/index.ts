import './styles/index.scss';

import * as Components from './components';
import { default as fetchFromServer, handleErrorResponse } from './utils/fetch';
import * as Constants from './constants';

export default {
  ...Components,
  Constants,
  fetchFromServer,
  handleErrorResponse
};
