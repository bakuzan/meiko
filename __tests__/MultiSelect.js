import React from 'react';
import { act } from '@testing-library/react';

import mockDocumentEventListeners from './__helpers__/documentEventListeners';

import { MultiSelect } from '../lib';

const { trigger } = mockDocumentEventListeners();

const mockedUpdateFn = jest.fn();

const options = [
  { value: 1, text: 'Gen1' },
  { value: 2, text: 'Gen2' },
  { value: 3, text: 'Gen3' },
  { value: 4, text: 'Gen4' }
];

const clickToOpenEvent = {
  type: 'click',
  stopPropagation: () => null
};

afterEach(() => jest.restoreAllMocks());

it('should render with minimum props', function () {
  const { container } = render(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  expect(container.firstChild).toBeTruthy();
  expect(container).toMatchSnapshot();
});

// todo must fix
xit('should open dropdown on input click', async function () {
  const { container, getByPlaceholderText } = render(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  const menuElement = document.querySelector(
    '.multi-select__dropdown-container'
  );

  expect(menuElement['aria-hidden']).toBe(true);
  expect(container).toMatchSnapshot();

  const user = userEvent.setup();
  await user.click(getByPlaceholderText('Select a jester'));

  expect(menuElement['aria-hidden']).toBe(false);
  expect(container).toMatchSnapshot();
});

it('should render correct number of options', async function () {
  const { container, getByPlaceholderText, getByLabelText } = render(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  const user = userEvent.setup();
  await user.click(getByPlaceholderText('Select a jester'));

  expect(getByLabelText('Select All')).toBeTruthy();
  options.forEach((option) => expect(getByLabelText(option.text)).toBeTruthy());

  expect(container).toMatchSnapshot();
});

// todo must fix
xit('should call onUpdate when select all clicked', async function () {
  const { container, getByPlaceholderText, getByLabelText } = render(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  const user = userEvent.setup();
  await user.click(getByPlaceholderText('Select a jester'));
  await user.click(getByLabelText('Select All'));

  expect(mockedUpdateFn).toHaveBeenCalledWith(
    options.map((x) => x.value),
    undefined
  );
  expect(container).toMatchSnapshot();
});

// todo must fix
xit('should call onUpdate when tickboxes are changed', async function () {
  const { container, getByLabelText, getByPlaceholderText, rerender } = render(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  const user = userEvent.setup();
  await user.click(getByPlaceholderText('Select a jester'));
  await user.click(getByLabelText(options[0].text));

  expect(mockedUpdateFn).toHaveBeenCalledWith([1], undefined);

  rerender(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[1]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  await user.click(getByLabelText(options[2].text));

  expect(mockedUpdateFn).toHaveBeenCalledWith([1, 4], undefined);
  expect(container).toMatchSnapshot();
});

xit('should close dropdown on outside click', async function () {
  const component = mount(
    <MultiSelect
      id="mko"
      placeholder="Select a jester"
      values={[]}
      options={options}
      onUpdate={mockedUpdateFn}
    />
  );

  await act(async () => {
    component.setState({ isOpen: true });
    trigger.click({ target: document.body });
  });

  const menuAfterOutsideClick = component.exists(
    '.multi-select__dropdown-container--is-open'
  );
  expect(menuAfterOutsideClick).toBe(false);
  expect(component).toMatchSnapshot();
  component.unmount();
});
