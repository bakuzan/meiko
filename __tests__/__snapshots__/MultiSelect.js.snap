// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onUpdate when select all clicked 1`] = `
<div>
  <div
    class="multi-select  mko_gty3bt-multiSelect"
  >
    <div
      class="multi-select__display has-float-label select-container  mko_gty3bt-multiSelect__display  mko_1aosyyv-controlContainer"
    >
      <input
        autocomplete="off"
        class="multi-select__input"
        id="mko"
        placeholder="Select a jester"
        readonly=""
        tabindex="0"
        type="text"
        value=""
      />
      <label
        for="mko"
      >
        MultiSelect
      </label>
    </div>
    <div
      aria-hidden="false"
      class="multi-select__dropdown-container multi-select__dropdown-container--is-open  mko_gty3bt-multiSelect__dropdown  mko_gty3bt-multiSelect__dropdown_open"
    >
      <ul
        class="multi-select__list  mko_gty3bt-multiSelect__list"
      >
        <input
          class="tab-trap tab-trap--top  mko_1q23t0y-tabTrap"
          type="text"
        />
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--selectAll"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--selectAll"
                name="mko--selectAll"
                type="checkbox"
              />
              Select All
            </label>
          </div>
        </li>
        <li
          class="multi-select__separator  mko_gty3bt-separator"
        />
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-0"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-0"
                name="mko--option-0"
                type="checkbox"
              />
              Gen1
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-1"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-1"
                name="mko--option-1"
                type="checkbox"
              />
              Gen2
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-2"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-2"
                name="mko--option-2"
                type="checkbox"
              />
              Gen3
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-3"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-3"
                name="mko--option-3"
                type="checkbox"
              />
              Gen4
            </label>
          </div>
        </li>
        <input
          class="tab-trap tab-trap--bottom  mko_1q23t0y-tabTrap"
          type="text"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`should call onUpdate when tickboxes are changed 1`] = `
<div
  className="multi-select  mko_gty3bt-multiSelect"
>
  <div
    className="multi-select__display has-float-label select-container  mko_gty3bt-multiSelect__display  mko_1aosyyv-controlContainer"
  >
    <input
      autoComplete="off"
      className="multi-select__input"
      id="mko"
      onClick={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a jester"
      readOnly={true}
      tabIndex={0}
      type="text"
      value="Gen1"
    />
    <label
      htmlFor="mko"
    >
      MultiSelect
    </label>
  </div>
  <div
    aria-hidden={false}
    className="multi-select__dropdown-container multi-select__dropdown-container--is-open  mko_gty3bt-multiSelect__dropdown  mko_gty3bt-multiSelect__dropdown_open"
  >
    <ForwardRef(TabTrap)
      className="multi-select__list  mko_gty3bt-multiSelect__list"
      element="ul"
      firstId="mko--selectAll"
      isActive={true}
      lastId="mko--option-3"
      onDeactivate={[Function]}
    >
      <li
        key="ALL"
      >
        <Tickbox
          checked={false}
          id="mko--selectAll"
          name="mko--selectAll"
          onChange={[Function]}
          text="Select All"
        />
      </li>
      <li
        className="multi-select__separator  mko_gty3bt-separator"
        key="SEPARATOR"
      />
      <li
        key="1"
      >
        <Tickbox
          checked={true}
          id="mko--option-0"
          name="mko--option-0"
          onChange={[Function]}
          text="Gen1"
        />
      </li>
      <li
        key="2"
      >
        <Tickbox
          checked={false}
          id="mko--option-1"
          name="mko--option-1"
          onChange={[Function]}
          text="Gen2"
        />
      </li>
      <li
        key="3"
      >
        <Tickbox
          checked={false}
          id="mko--option-2"
          name="mko--option-2"
          onChange={[Function]}
          text="Gen3"
        />
      </li>
      <li
        key="4"
      >
        <Tickbox
          checked={false}
          id="mko--option-3"
          name="mko--option-3"
          onChange={[Function]}
          text="Gen4"
        />
      </li>
    </ForwardRef(TabTrap)>
  </div>
</div>
`;

exports[`should open dropdown on input click 1`] = `
<div
  className="multi-select  mko_gty3bt-multiSelect"
>
  <div
    className="multi-select__display has-float-label select-container  mko_gty3bt-multiSelect__display  mko_1aosyyv-controlContainer"
  >
    <input
      autoComplete="off"
      className="multi-select__input"
      id="mko"
      onClick={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a jester"
      readOnly={true}
      tabIndex={0}
      type="text"
      value=""
    />
    <label
      htmlFor="mko"
    >
      MultiSelect
    </label>
  </div>
  <div
    aria-hidden={false}
    className="multi-select__dropdown-container multi-select__dropdown-container--is-open  mko_gty3bt-multiSelect__dropdown  mko_gty3bt-multiSelect__dropdown_open"
  >
    <ForwardRef(TabTrap)
      className="multi-select__list  mko_gty3bt-multiSelect__list"
      element="ul"
      firstId="mko--selectAll"
      isActive={true}
      lastId="mko--option-3"
      onDeactivate={[Function]}
    >
      <li
        key="ALL"
      >
        <Tickbox
          checked={false}
          id="mko--selectAll"
          name="mko--selectAll"
          onChange={[Function]}
          text="Select All"
        />
      </li>
      <li
        className="multi-select__separator  mko_gty3bt-separator"
        key="SEPARATOR"
      />
      <li
        key="1"
      >
        <Tickbox
          checked={false}
          id="mko--option-0"
          name="mko--option-0"
          onChange={[Function]}
          text="Gen1"
        />
      </li>
      <li
        key="2"
      >
        <Tickbox
          checked={false}
          id="mko--option-1"
          name="mko--option-1"
          onChange={[Function]}
          text="Gen2"
        />
      </li>
      <li
        key="3"
      >
        <Tickbox
          checked={false}
          id="mko--option-2"
          name="mko--option-2"
          onChange={[Function]}
          text="Gen3"
        />
      </li>
      <li
        key="4"
      >
        <Tickbox
          checked={false}
          id="mko--option-3"
          name="mko--option-3"
          onChange={[Function]}
          text="Gen4"
        />
      </li>
    </ForwardRef(TabTrap)>
  </div>
</div>
`;

exports[`should render correct number of options 1`] = `
<div>
  <div
    class="multi-select  mko_gty3bt-multiSelect"
  >
    <div
      class="multi-select__display has-float-label select-container  mko_gty3bt-multiSelect__display  mko_1aosyyv-controlContainer"
    >
      <input
        autocomplete="off"
        class="multi-select__input"
        id="mko"
        placeholder="Select a jester"
        readonly=""
        tabindex="0"
        type="text"
        value=""
      />
      <label
        for="mko"
      >
        MultiSelect
      </label>
    </div>
    <div
      aria-hidden="false"
      class="multi-select__dropdown-container multi-select__dropdown-container--is-open  mko_gty3bt-multiSelect__dropdown  mko_gty3bt-multiSelect__dropdown_open"
    >
      <ul
        class="multi-select__list  mko_gty3bt-multiSelect__list"
      >
        <input
          class="tab-trap tab-trap--top  mko_1q23t0y-tabTrap"
          type="text"
        />
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--selectAll"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--selectAll"
                name="mko--selectAll"
                type="checkbox"
              />
              Select All
            </label>
          </div>
        </li>
        <li
          class="multi-select__separator  mko_gty3bt-separator"
        />
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-0"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-0"
                name="mko--option-0"
                type="checkbox"
              />
              Gen1
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-1"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-1"
                name="mko--option-1"
                type="checkbox"
              />
              Gen2
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-2"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-2"
                name="mko--option-2"
                type="checkbox"
              />
              Gen3
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-3"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-3"
                name="mko--option-3"
                type="checkbox"
              />
              Gen4
            </label>
          </div>
        </li>
        <input
          class="tab-trap tab-trap--bottom  mko_1q23t0y-tabTrap"
          type="text"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`should render with minimum props 1`] = `
<div>
  <div
    class="multi-select  mko_gty3bt-multiSelect"
  >
    <div
      class="multi-select__display has-float-label select-container  mko_gty3bt-multiSelect__display  mko_1aosyyv-controlContainer"
    >
      <input
        autocomplete="off"
        class="multi-select__input"
        id="mko"
        placeholder="Select a jester"
        readonly=""
        tabindex="0"
        type="text"
        value=""
      />
      <label
        for="mko"
      >
        MultiSelect
      </label>
    </div>
    <div
      aria-hidden="true"
      class="multi-select__dropdown-container  mko_gty3bt-multiSelect__dropdown"
    >
      <ul
        class="multi-select__list  mko_gty3bt-multiSelect__list"
      >
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--selectAll"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--selectAll"
                name="mko--selectAll"
                type="checkbox"
              />
              Select All
            </label>
          </div>
        </li>
        <li
          class="multi-select__separator  mko_gty3bt-separator"
        />
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-0"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-0"
                name="mko--option-0"
                type="checkbox"
              />
              Gen1
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-1"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-1"
                name="mko--option-1"
                type="checkbox"
              />
              Gen2
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-2"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-2"
                name="mko--option-2"
                type="checkbox"
              />
              Gen3
            </label>
          </div>
        </li>
        <li>
          <div
            class="tickbox  mko_1oj1nfd-tickbox"
          >
            <label
              class="tickbox__label  mko_1oj1nfd-tickbox__label"
              for="mko--option-3"
            >
              <input
                class=" mko_1oj1nfd-tickbox__input"
                id="mko--option-3"
                name="mko--option-3"
                type="checkbox"
              />
              Gen4
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
